security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -968,6 +984,10 @@ func (sc *SecretCache) sendRetriableRequest(ctx context.Context, csrPEM []byte,
 		var httpRespCode int
 		if isCSR {
 			requestErrorString = fmt.Sprintf("%s CSR", logPrefix)
+			if !withToken {
```

Comment:
I think the logic with 'withToken' and all the extra params is too complex: all you need is:
- if certs exist, use them
- else if token present and not expired - use token
- else fail

That does the same thing, and it's what we want.

Reply:
Ok good idea I think it is a good idea here

Reply:
Add it in our logic. more simple